From dbba99a383b545f7ec70bc385e760b5a4ca10a13 Mon Sep 17 00:00:00 2001
From: Andre Mauricio Zelak <andre.zelak@windriver.com>
Date: Wed, 29 Apr 2026 19:15:15 -0300
Subject: [PATCH] Extend leap seconds expire date until 28 December 2036

Custom leap-seconds.list file to extend expire date until
28 December 2036.

Derived from IERS Bulletin C 72 (2026-07-06)
https://hpiers.obspm.fr/iers/bul/bulc/bulletinc.72

* leap-seconds.list retrieve from:
https://data.iana.org/time-zones/tzdb-2026c/leap-seconds.list

Signed-off-by: Andre Mauricio Zelak <andre.zelak@windriver.com>
[cr: Refresh patch for 2026c]
Signed-off-by: Viorel-Catalin Rapiteanu <viorel-catalin.rapiteanu@windriver.com>
---
 leap-seconds.list | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/leap-seconds.list b/leap-seconds.list
index 0b5b000..5e400c8 100644
--- a/leap-seconds.list
+++ b/leap-seconds.list
@@ -1,3 +1,12 @@
+#	This file was generated by Wind River Systems, and it is intended for use
+#	in StarlingX systems only.
+#
+#	The content is derived from the IANA Time Zone Database version 2026c, and
+#	its expire date was extended to 28 December 2036. The original file can be
+#	downloaded from the following link:
+#
+#	https://data.iana.org/time-zones/tzdb-2026c/leap-seconds.list
+#
 #	ATOMIC TIME
 #	Coordinated Universal Time (UTC) is the reference time scale derived
 #	from The "Temps Atomique International" (TAI) calculated by the Bureau
@@ -56,19 +65,20 @@
 #
 #	1) Last update of the file.
 #
-#	Updated through IERS Bulletin C (https://hpiers.obspm.fr/iers/bul/bulc/bulletinc.dat)
+#	Updated on 28 September 2026 by Wind River Systems as part of StarlingX project, derived from
+#	IERS Bulletin C 72 of 06 July 2026 (https://hpiers.obspm.fr/iers/bul/bulc/bulletinc.dat).
 #
 #	The following line shows the last update of this file in NTP timestamp:
 #
-#$	3992312697
+#$	3999542400
 #
 #	2) Expiration date of the file given on a semi-annual basis: last June or last December
 #
-#	File expires on 28 June 2027
+#	File expires on 28 December 2036
 #
 #	Expire date in NTP timestamp:
 #
-#@	4023129600
+#@	4323024000
 #
 #
 #	LIST OF LEAP SECONDS
@@ -117,4 +127,4 @@
 #	please see the readme file in the 'source' directory :
 #	https://hpiers.obspm.fr/iers/bul/bulc/ntp/sources/README
 #
-#h	a9bad145 84c31c70 758402aa b37bfd54 5923836a
+#h	0b4608d1 502bf906 e55dee07 0772b433 d181c82c
-- 
2.47.3
